go/private/actions/compilepkg.bzl

```diff
@@ -108,10 +124,13 @@ def emit_compilepkg(
         args.add("-p", importmap)
     args.add("-package_list", go.package_list)
 
-    args.add("-o", out_lib)
-    args.add("-x", out_export)
-    nogo = go.get_nogo(go)
+    args.add("-lo", out_lib)
+    args.add("-o", out_export)
```

Comment:
Just to make sure we are on the same page:

- `go tool compile -o` should write to `*.a` or `*.o` files. If `-pack` is set, which we do set it unconditionally, then it's always `*.a` file.

- `go tool compile -linkobj` should write to `*.lo` files.

This seems to be mapping it in reverse(?):

- `-lo` -> out_lib -> `.a`
- `-o` -> out_export -> `.x`

There is not a strong precedent for these, and the extension suppose to change based on usage of `-pack` and `-linkobj`.

https://cs.opensource.google/search?q=%22-linkobj%22&ss=go

I suggest we stick to the example provided in https://cs.opensource.google/go/go/+/master:test/linkobj.go;l=75-90;drc=b2fd76ab8dd72b9f2135c1e9b15130e6028fc838

```bash
go tool compile ... -pack=1 ... -o p1.a -linkobj p1.lo
```

WDYT?

Reply:
Re-reading https://go-review.googlesource.com/c/go/+/22500

> The first part is for reading by future compiles.
The second part is for reading by the final link step.
Splitting the file this way allows distributed build systems
to ship the compile-input part only to compile steps and
the linker-input part only to linker steps.

The implication here is that for subsequent compile actions, we should only need `.a`. And for downstream link actions, we should only need `.lo`.

Right now, we are including both output types in both compile and link actions?
Potentially, this could be split into different providers for downstream action to consume separately, making the input root trees for those actions relatively smaller.
No need to change it in this PR though.

Reply:
Hmm now im not sure if the linker requires the link object file to be ".a" for it to work 🤔

Reply:
> Right now, we are including both output types in both compile and link actions?

The compilepkg action only depends on the export files of its dependencies, so we should already be doing this right.

Reply:
Perhaps dropbox’s rules could be a good point of reference 

https://github.com/dropbox/dbx_build_tools/blob/ee7d614c85217adb061162ed0e5fd852d893b939/build_tools/go/go.bzl#L426-L431

Reply:
> Hmm now im not sure if the linker requires the link object file to be ".a" for it to work 🤔

That's what I'm thinking as well: the full link object may be handed over to external linkers not under our control, so I would say it's safer to keep the standard `.a` extension.

We can always rename the files in a separate change to improve consistency with vanilla Go.

Reply:
> The compilepkg action only depends on the export files of its dependencies, so we should already be doing this right.

I will test this PR out and double check this. I think you might be right though.